oauthproxy.go

```diff
@@ -797,6 +816,172 @@ func (p *OAuthProxy) backendLogout(rw http.ResponseWriter, req *http.Request) {
+func (p *OAuthProxy) frontendLogout(rw http.ResponseWriter, req *http.Request, session *sessionsapi.SessionState, fallbackRedirect string) {
+	// Try to get the logout URL (cached or discovered)
+	logoutURL := p.getLogoutURL()
+	if logoutURL == "" {
+		// Fallback: simple redirect if no logout URL available
+		logger.Printf("SignOut: No logout URL available, using fallback redirect")
+		http.Redirect(rw, req, fallbackRedirect, http.StatusFound)
+		return
+	}
+
+	// Add id_token_hint if session and token are available
+	if session != nil && session.IDToken != "" {
+		separator := "?"
+		if strings.Contains(logoutURL, "?") {
+			separator = "&"
+		}
+		logoutURL += separator + "id_token_hint=" + url.QueryEscape(session.IDToken)
+		logger.Printf("SignOut: Redirecting to OIDC logout with id_token_hint")
+	} else {
+		logger.Printf("SignOut: Redirecting to OIDC logout without id_token_hint")
+	}
+
+	logger.Printf("SignOut: Using logout URL: %s", redactSensitiveQueryParams(logoutURL))
+	http.Redirect(rw, req, logoutURL, http.StatusFound)
+}
+
+// discoverLogoutURL fetches the logout URL from OIDC .well-known endpoint
+func (p *OAuthProxy) discoverLogoutURL() (string, error) {
+	providerData := p.provider.Data()
+
+	// Validate that LoginURL is available and properly formatted
+	if providerData.LoginURL == nil {
+		return "", fmt.Errorf("could not determine issuer URL for OIDC discovery")
+	}
+
+	loginURL := providerData.LoginURL
+
+	// Ensure LoginURL has HTTPS scheme and a host for security
+	if loginURL.Scheme != "https" {
+		return "", fmt.Errorf("could not determine issuer URL for OIDC discovery")
+	}
+	if loginURL.Host == "" {
+		return "", fmt.Errorf("could not determine issuer URL for OIDC discovery")
+	}
+
+	// Derive issuer URL by parsing and manipulating the login URL
+	// For Keycloak OIDC: https://host/realms/realm/protocol/openid-connect/auth -> https://host/realms/realm
+	issuerURL := &url.URL{
+		Scheme: loginURL.Scheme,
+		Host:   loginURL.Host,
+	}
+
+	// Trim known OIDC path segments to get the issuer
+	loginPath := loginURL.Path
+	if strings.HasSuffix(loginPath, "/protocol/openid-connect/auth") {
+		issuerURL.Path = strings.TrimSuffix(loginPath, "/protocol/openid-connect/auth")
+	} else {
+		return "", fmt.Errorf("could not determine issuer URL for OIDC discovery")
+	}
+
+	// Construct .well-known URL using proper URL methods
+	wellKnownPath := strings.TrimSuffix(issuerURL.Path, "/") + "/.well-known/openid-configuration"
+	wellKnownURL := &url.URL{
+		Scheme: issuerURL.Scheme,
+		Host:   issuerURL.Host,
+		Path:   wellKnownPath,
+	}
+
+	// Verify well-known URL host matches login URL host to prevent SSRF
+	if wellKnownURL.Host != loginURL.Host {
+		return "", fmt.Errorf("could not determine issuer URL for OIDC discovery")
+	}
+
+	// Create HTTP client with timeout to prevent hanging requests
+	client := &http.Client{
+		Timeout: 10 * time.Second,
```

Comment:
Is 10 seconds enough for external network call?
I suggest bumping to 30 seconds.